nomos-services/data-availability/network/src/membership/adapters/mock.rs

```diff
@@ -58,21 +61,21 @@ where
         }
     }
 
-    async fn update(&self, block_number: u64, new_members: HashMap<PeerId, Multiaddr>) {
-        let updated_membership = self.handler.membership().update(new_members);
-        let assignations = updated_membership.subnetworks();
-
-        self.handler.update(updated_membership);
-        self.storage.store(block_number, assignations);
-    }
-
     async fn get_historic_membership(&self, block_number: u64) -> Option<Membership> {
         let assignations = self.storage.get(block_number)?;
         Some(self.handler.membership().init(assignations))
     }
 
-    async fn subscribe(&self) -> Result<MembershipSnapshotStream, MembershipAdapterError> {
-        self.subscribe_stream(ServiceType::DataAvailability).await
+    async fn bootstrap(self: Arc<Self>) -> Result<(), MembershipAdapterError> {
+        let mut stream = self.subscribe_stream(ServiceType::DataAvailability).await?;
+
+        tokio::spawn(async move {
+            while let Some((block_number, new_members)) = stream.next().await {
+                Self::update(block_number, new_members);
+            }
+        });
+
+        Ok(())
```

Comment:
This is a sign of bad design. We used to have a stream which is polled by the upper service, there is no need for more tasks. We should avoid creating this tiny tasks.